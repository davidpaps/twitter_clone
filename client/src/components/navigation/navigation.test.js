import React from "react";
import { shallow } from "enzyme";
import Navigation from "./navigation";

describe("Navigation", () => {
  let wrapper;

  beforeEach(() => (wrapper = shallow(<Navigation />)));

  it("should render a <Fragment />", () => {
    expect(wrapper.find("Fragment").length).toEqual(1);
  });

  it("should render a <div />", () => {
    expect(wrapper.find("div").length).toEqual(3);
  });

  it("should render a <button />", () => {
    expect(wrapper.find("button").length).toEqual(1);
  });

  it("should render a <ul />", () => {
    expect(wrapper.find("ul").length).toEqual(1);
  });

  it("should render a <li />", () => {
    expect(wrapper.find("li").length).toEqual(1);
  });
});
