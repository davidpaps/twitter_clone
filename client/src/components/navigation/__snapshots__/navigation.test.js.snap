// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render correctly 1`] = `
<div
  className="Navigation"
>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <img
        alt=""
        className="d-inline-block align-top"
        height="30"
        loading="lazy"
        src="https://vignette.wikia.nocookie.net/logopedia/images/9/99/Twitter_2012.svg/revision/latest/scale-to-width-down/340?cb=20180321032720"
        width="30"
      />
      Twitter
    </a>
    <div
      className="navbar-collapse collapse"
    />
    <div>
      <ul
        className="nav navbar-nav navbar-center"
      >
        <li />
      </ul>
    </div>
    <div>
      <ul
        className="nav navbar-nav navbar-right"
      >
        <li
          className="nav-item active"
        >
          <a
            className="nav-link"
            href="/sign_up"
          >
            Sign In 
            <span
              className="sr-only"
            >
              (current)
            </span>
            <img
              alt=""
              className="d-inline-block "
              height="20"
              loading="lazy"
              src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcTg9NgXjYkbVD1Wb4LzGSv9nYR6bC3Z_B4qWw&usqp=CAU"
              width="20"
            />
          </a>
        </li>
        <li
          className="nav-item active"
        >
          <a
            className="nav-link"
            href="/sign_in"
          >
            Sign Up 
            <span
              className="sr-only"
            >
              (current)
            </span>
            <img
              alt=""
              className="d-inline-block"
              height="20"
              loading="lazy"
              src="https://cdn3.iconfinder.com/data/icons/user-interface-2-9/34/169-512.png"
              width="20"
            />
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
